src/types/qwxwayland.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2023 Dingyuan Zhang <zhangdingyuan@uniontech.com>.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+
+#include "qwxwayland.h"
+
+#include <qwdisplay.h>
+#include <qwcompositor.h>
+#include <qwseat.h>
+#include <QImage>
+#include <QPoint>
+
+extern "C" {
+#include <math.h>
+#define class _class
+#include <wlr/xwayland/xwayland.h>
+#undef class
+}
+
+QW_BEGIN_NAMESPACE
+
+QWXWayland *QWXWayland::create(QWDisplay *wl_display, QWCompositor *compositor, bool lazy)
+{
+    auto* pointer = wlr_xwayland_create(wl_display->handle(), compositor->handle(), lazy);
+    return pointer ? from(pointer) : nullptr;
+}
+
+QWXWayland* QWXWayland::from(wlr_xwayland* xwayland)
+{
+    return reinterpret_cast<QWXWayland*>(xwayland);
+}
+
+void QWXWayland::destroy()
+{
+    wlr_xwayland_destroy(handle());
+}
+
+void QWXWayland::setCursor(QImage image, const QPoint& hotspot)
```

Comment:
`QImage img = image.convertToFormat(QImage::Format_ARGB32_Premultiplied);` before using the QImage.